crates/eth-rpc/tests/integration.rs

```diff
@@ -1,31 +1,136 @@
 #[cfg(test)]
 mod integration_tests {
-    use std::str::FromStr;
+    use std::convert::TryFrom;
+    use std::sync::Arc;
+    use std::time::Duration;
 
-    use ethers::prelude::{Block as EthersBlock, Http as EthersHttp, H256 as EthersH256};
+    use ethers::contract::ContractFactory;
+    use ethers::core::k256::ecdsa::SigningKey;
+    use ethers::middleware::SignerMiddleware;
+    use ethers::prelude::abigen;
+    use ethers::providers::{Http, Middleware, Provider};
+    use ethers::signers::{LocalWallet, Signer};
+    use ethers::types::{BlockId, BlockNumber, TransactionReceipt, H160, H256};
+    use ethers::utils::keccak256;
+    use hex::FromHex;
+    use kakarot_test_utils::execution::eoa::EOA;
     use kakarot_test_utils::fixtures::katana;
     use kakarot_test_utils::rpc::start_kakarot_rpc_server;
     use kakarot_test_utils::sequencer::Katana;
     use reth_primitives::U64;
     use rstest::*;
 
+    abigen!(ERC20, "tests/contracts/ERC20/IERC20.json");
+
+    // ⚠️ Only one test with a Katana fixture can run at a time.
+    // When trying to run two tests with a Katana fixture, the second test will fail with:
+    // `thread 'test_erc20' panicked at 'Failed to start the server: Os { code: 98, kind:
+    // AddrInUse, message: "Address already in use" }'`
     #[rstest]
     #[awt]
     #[tokio::test(flavor = "multi_thread")]
-    async fn test_get_block_rpc(#[future] katana: Katana) {
-        // Start the Kakarot RPC server
+    async fn test_erc20(#[future] katana: Katana) {
```

Comment:
I find that this function contains a lot of asserts, could we perhaps have the expected return inside a json file and just deserialize the content and compare?

Reply:
what is the advantage of such an approach?